src/v/kafka/server/group.cc

```diff
@@ -728,6 +730,7 @@ void group::try_prepare_rebalance() {
     auto prev_state = set_state(group_state::preparing_rebalance);
 
     if (prev_state == group_state::empty) {
+        _initial_join_in_progress = true;
         // debounce joins to an empty group. for a bounded delay, we'll avoid
         // competing the join phase as long as new members are arriving.
         auto rebalance = rebalance_timeout();
```

Comment:
I was looking at this as well i am wondering if this code would ever be called ? After one of the members joining fiber will call `try_prepare_rebalance` it will always exit early in `if (!valid_previous_state(group_state::preparing_rebalance)) {`